feat: user can find ipdc products by typing

JonasVanHoof opened the pull request:
## Description

Users are only able to fill in a number for finding ipdc products. It would be usefull to also be able to type text in that field to find the products.

## How to test

1. Go to a editable process
2. Look for ipdc products
3. Results should be shown
4. You should be able to safe these product types

JonasVanHoof committed:
feat: user can find ipdc products by typing

The "zoekterm" param is returning weird results AFAIK

Refs: oph-691

MartijnBogaert reviewed (changes requested):
Is it possible to tell the IPDC API to only match the search param to IPDC titles? Because now it looks like it searches everywhere in the IPDC product data, giving unexpected results like this:

<img width="654" height="313" alt="Image" src="https://github.com/user-attachments/assets/c571f4a9-6948-49d9-afbb-671e4e0e5f6f" />

JonasVanHoof committed:
feat: add extra explicit queryParam to only search on non archived

This param makes sure we only show result for products that have not been archived yet

Refs: oph-691

JonasVanHoof committed:
hack: only show the results where the title includes the searchparam for ipdc

The api does not support to only search for the title content. This means that we have to do this frontend hack

Refs: oph-691

MartijnBogaert commented on app/components/process/details-card/ipdc-multiple-select.js:
```diff
@@ -11,29 +11,52 @@ export default class ProcessDetailsCardIpdcMultipleSelectComponent extends Compo
+      results = [jsonResponse];
+    } else {
+      const response = yield fetch(
+        `/ipdc/doc/product?zoekterm=${searchParams}`,
```
We'll leave it as is then! Once it's deployed on dev, we should not forget to ask Stefanie to try it out and give her opinion on the UX.

MartijnBogaert commented on app/components/process/details-card/ipdc-multiple-select.js:
```diff
@@ -11,29 +11,52 @@ export default class ProcessDetailsCardIpdcMultipleSelectComponent extends Compo
+      results = [jsonResponse];
+    } else {
+      const response = yield fetch(
+        `/ipdc/doc/product?zoekterm=${searchParams}`,
```
Oh I see now that you did this is in https://github.com/lblod/frontend-openproceshuis/pull/137. Gonna resove and approve so this can be merged and finished in the other PR.
